utility_analysis_new/combiners.py

```diff
@@ -18,6 +18,7 @@
 import numpy as np
 import math
 import scipy
+from numpy import ndarray
```

Comment:
Google Python coding style, import modules, not classes
Nit: remove this import and use `np.ndarray`